server/src/infra/repositories/person.repository.ts

```diff
@@ -84,6 +83,8 @@ export class PersonRepository implements IPersonRepository {
       .limit(500);
     if (!options?.withHidden) {
       queryBuilder.andWhere('person.isHidden = false');
+    } else {
+      queryBuilder.orderBy('person.isHidden', 'ASC');
```

Comment:
What do you want to do ?
With this change, people with names are not prioritized to be at the top of the list.

Reply:
If `withHidden` is set to true, all entries returned will have `isHidden` set to `FALSE` and thus sorting by `isHidden` does not change the returned items.

If I'm not mistaken, then the following two queries should be the same (with `withHidden = true`):
```sql
SELECT
  "person"."id" AS "person_id",
  "person"."createdAt" AS "person_createdAt",
  "person"."updatedAt" AS "person_updatedAt",
  "person"."ownerId" AS "person_ownerId",
  "person"."name" AS "person_name",
  "person"."birthDate" AS "person_birthDate",
  "person"."thumbnailPath" AS "person_thumbnailPath",
  "person"."faceAssetId" AS "person_faceAssetId",
  "person"."isHidden" AS "person_isHidden"
FROM "person" "person"
LEFT JOIN "asset_faces" "face"
  ON "face"."personId"="person"."id"
INNER JOIN "assets" "asset"
  ON "asset"."id"="face"."assetId"
    AND ("asset"."deletedAt" IS NULL)
WHERE
  "person"."ownerId" = $1
  AND "person"."isHidden" = FALSE
GROUP BY "person"."id"
HAVING
  "person"."name" != ''
  OR COUNT("face"."assetId") >= $2
ORDER BY
  "person"."isHidden" ASC,
  NULLIF("person"."name", '') IS NULL ASC,
  COUNT("face"."assetId") DESC,
  NULLIF("person"."name", '') ASC NULLS LAST
LIMIT 500
```
and 
```sql
SELECT
  "person"."id" AS "person_id",
  "person"."createdAt" AS "person_createdAt",
  "person"."updatedAt" AS "person_updatedAt",
  "person"."ownerId" AS "person_ownerId",
  "person"."name" AS "person_name",
  "person"."birthDate" AS "person_birthDate",
  "person"."thumbnailPath" AS "person_thumbnailPath",
  "person"."faceAssetId" AS "person_faceAssetId",
  "person"."isHidden" AS "person_isHidden"
FROM "person" "person"
LEFT JOIN "asset_faces" "face"
  ON "face"."personId"="person"."id"
INNER JOIN "assets" "asset"
  ON "asset"."id"="face"."assetId"
    AND ("asset"."deletedAt" IS NULL)
WHERE
  "person"."ownerId" = $1
  AND "person"."isHidden" = FALSE
GROUP BY "person"."id"
HAVING
  "person"."name" != ''
  OR COUNT("face"."assetId") >= $2
ORDER BY
  NULLIF("person"."name", '') IS NULL ASC,
  COUNT("face"."assetId") DESC,
  NULLIF("person"."name", '') ASC NULLS LAST
LIMIT 500
```

Reply:
No, this change the sql query : 
original :
```sql
SELECT
    "person"."id" AS "person_id",
    "person"."createdAt" AS "person_createdAt",
    "person"."updatedAt" AS "person_updatedAt",
    "person"."ownerId" AS "person_ownerId",
    "person"."name" AS "person_name",
    "person"."birthDate" AS "person_birthDate",
    "person"."thumbnailPath" AS "person_thumbnailPath",
    "person"."faceAssetId" AS "person_faceAssetId",
    "person"."isHidden" AS "person_isHidden"
FROM
    "person" "person"
LEFT JOIN
    "asset_faces" "face" ON "face"."personId" = "person"."id"
INNER JOIN
    "assets" "asset" ON "asset"."id" = "face"."assetId" AND ("asset"."deletedAt" IS NULL)
WHERE
    "person"."ownerId" = $1
GROUP BY
    "person"."id"
HAVING
    "person"."name" != '' OR COUNT("face"."assetId") >= $2
ORDER BY
    "person"."isHidden" ASC,
    NULLIF("person"."name", '') IS NULL ASC,
    COUNT("face"."assetId") DESC,
    NULLIF("person"."name", '') ASC NULLS LAST
LIMIT
    500;
```

your changes :
```sql
SELECT
    "person"."id" AS "person_id",
    "person"."createdAt" AS "person_createdAt",
    "person"."updatedAt" AS "person_updatedAt",
    "person"."ownerId" AS "person_ownerId",
    "person"."name" AS "person_name",
    "person"."birthDate" AS "person_birthDate",
    "person"."thumbnailPath" AS "person_thumbnailPath",
    "person"."faceAssetId" AS "person_faceAssetId",
    "person"."isHidden" AS "person_isHidden"
FROM
    "person" "person"
LEFT JOIN
    "asset_faces" "face" ON "face"."personId" = "person"."id"
INNER JOIN
    "assets" "asset" ON "asset"."id" = "face"."assetId" AND ("asset"."deletedAt" IS NULL)
WHERE
    "person"."ownerId" = $1
GROUP BY
    "person"."id"
HAVING
    "person"."name" != '' OR COUNT("face"."assetId") >= $2
ORDER BY
    "person"."isHidden" ASC
LIMIT 500;
```

Reply:
Ah, I now understand why: `addOrderBy` appends entries to an `ORDER BY`.

I'm sorry for the confusion – I've reverted this change.

Reply:
Correct me if I'm wrong but you always want the hidden people to come after the non-hidden people. With your changes and `withHidden` set to false, if someone has more than 500 people hidden, the query will return nothing.